### ISSUE TITLE
NULL constant in PreparedStatement

### ISSUE THREAD

**Issue by bgranvea** [ClickHouse/clickhouse-jdbc#495]
it seems to be related to #261.

I cannot use NULL constant in PreparedStatement, I get an error "Cannot parse expression of type Nullable(String) here: \N)"

Test case:

```
create table test (str Nullable(String)) ENGINE=MergeTree ORDER BY tuple();

@Test
public void testRequest() throws SQLException {
    try (Connection c = DriverManager.getConnection("jdbc:clickhouse://localhost:8123", "xxxx", "xxxx") {
        // works
        Statement stmt = c.createStatement();
        stmt.executeUpdate("INSERT INTO test(str) VALUES (null)");

        // doesn't work
        PreparedStatement ps = c.prepareStatement("INSERT INTO test(str) VALUES (null)");
        ps.executeUpdate();
    }
}
```

Here is what is sent to the server:

```
2020/09/29 10:01:09:724 CEST [DEBUG] wire - http-outgoing-0 >> "POST /?compress=1&query_id=26eca5df-cdcc-44cb-8c37-e62291f7d835&database=default&password=xxxx&user=xxxx HTTP/1.1[\r][\n]"
2020/09/29 10:01:09:725 CEST [DEBUG] wire - http-outgoing-0 >> "Content-Length: 33[\r][\n]"
2020/09/29 10:01:09:725 CEST [DEBUG] wire - http-outgoing-0 >> "Content-Type: text/plain; charset=UTF-8[\r][\n]"
2020/09/29 10:01:09:725 CEST [DEBUG] wire - http-outgoing-0 >> "Host: localhost:8123[\r][\n]"
2020/09/29 10:01:09:725 CEST [DEBUG] wire - http-outgoing-0 >> "Connection: Keep-Alive[\r][\n]"
2020/09/29 10:01:09:726 CEST [DEBUG] wire - http-outgoing-0 >> "User-Agent: Apache-HttpClient/4.5.3 (Java/1.8.0_201)[\r][\n]"
2020/09/29 10:01:09:726 CEST [DEBUG] wire - http-outgoing-0 >> "[\r][\n]"
2020/09/29 10:01:09:726 CEST [DEBUG] wire - http-outgoing-0 >> "INSERT INTO test(str) VALUES (\N)"
2020/09/29 10:01:09:764 CEST [DEBUG] wire - http-outgoing-0 << "HTTP/1.1 400 Bad Request[\r][\n]"

```
Tested with JDBC driver 0.2.4.

**Comment by enqueue** [comment]
Thanks for the detailed description and the test case! I had not thought of this use, and will try to fix it.

Meanwhile, you could use the "canonical" way :smile: :
```
PreparedStatement ps = c.prepareStatement("INSERT INTO test(str) VALUES (?)");
ps.setNull(1, Types.VARCHAR);
ps.executeUpdate();
```

**Comment by zhicwu** [comment]
Fixed on `develop` branch and will be merged into `master` after 0.2.5 release this weekend.
